### PR TITLE
Support multi-line descriptions

### LINKED ISSUE #47: Error in the pyproject.toml file
It looks like there is a spacing issue in the default description value for the pyproject toml. 

```
description = "A project that adds numbers.
"
```

please note that i haven't tested this to see if it's something that I did when i entered the description or something native to the template!

### PR DESCRIPTION
**Pull request by Midnighter**
Fix #47

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -63,11 +63,12 @@ dependencies = [
...
+    "pytest-xdist ~=3.6",
     "pre-commit"
 ]
 
 [[tool.hatch.envs.test.matrix]]
 python = ["3.10", "3.11", "3.12", "3.13"]
 
 [tool.hatch.envs.test.scripts]
-run = "pytest {args}"
+run = "pytest {args:--numprocesses=auto}"
```
Comment (blink1073) [inline pyproject.toml:74]: Was this change intentional?  I've often had cases where I couldn't use pytest-xdist.

Comment (Midnighter) [reply pyproject.toml:74]: It was, as I was waiting longer than desired for the tests to finish. To be honest, it's besides the point of the issue, so I should have made it separate.

Comment (Midnighter) [reply pyproject.toml]: Okay, I've removed those changes here and made a separate PR https://github.com/pyOpenSci/pyos-package-template/pull/55

**Comment by sneakers-the-rat** [comment]
oh wow how did i miss that there was already a PR for this.

maybe we can combine the single/multiline siwtching from mine and the nice wrapping from yours?

https://github.com/pyOpenSci/pyos-package-template/pull/53

**Comment by Midnighter** [comment]
> maybe we can combine the single/multiline siwtching from mine and the nice wrapping from yours?

How about this one?

```jinja
{% set wrapped_description = package_description | trim | wordwrap(width=88) %}
description = {% if "\n" in wrapped_description -%}
"""
{{ wrapped_description }}
"""
{%- else -%}
"{{ wrapped_description }}"
{%- endif %}
```

Possibly with the backslashes in the multi-line string. Although I fear that may cause confusion without bringing huge benefit.

**Comment by sneakers-the-rat** [comment]
aha i guess toml automatically strips the leading whitespace. so 

```toml
description = """
hey
"""
```

parses as

```
'hey\n'
```

imo ya the leading newline is the more annoying one, i don't think it's a big deal to have a trailing newline on the description or not so sure i'm good with leaving out the backslash.

We probably want to strip the whitespace tho, bc that version^ renders like:

```toml
[project]
name = "mine"

version = "0.1.0"


description = "single line description"
authors = [
    { name = "me", email = "me" },
]
```

and

```toml
[project]
name = "mine"

version = "0.1.0"


description = """
multiline

description

hello
"""
authors = [
    { name = "me", email = "me" },
]
```

So how about adding the whitespace gulper before the `set` like:

```jinja
{%- set wrapped_description = package_description | trim | wordwrap(width=88) %}
description = {% if "\n" in wrapped_description -%}
"""
{{ wrapped_description }}
"""
{%- else -%}
"{{ wrapped_description }}"
{%- endif %}
```

and I pr'd already for the rest of the whitespace :) https://github.com/pyOpenSci/pyos-package-template/pull/56

**Comment by Midnighter** [comment]
I stopped caring about aligning whitespace, but yes, that seems good to me.

**Comment by sneakers-the-rat** [comment]
> I stopped caring about aligning whitespace

ha i basically always reach that point and am just like "heck it i'll just use a formatter after generation" but until that point my eyes cannot look away. it is a personality flaw

**Review by sneakers-the-rat — APPROVED** [review]
locks god toe mea 👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍
